src/lib/tls/tls_session.h

```diff
@@ -248,8 +248,29 @@ class BOTAN_PUBLIC_API(3, 0) Session_Summary : public Session_Base {
        */
       const std::optional<Session_Ticket>& session_ticket() const { return m_session_ticket; }
 
-      bool psk_used() const { return m_psk_used; }
+      /**
+       * The negotiated identity of an externally provided preshared key used to
+       * establish this session. For TLS 1.3 this may be any of the externally
+       * provided PSKs offered by the client. PSK identities used as session
+       * tickets for TLS 1.3 session resumption won't be shown here.
+       */
+      const std::optional<std::vector<uint8_t>>& external_psk_identity() const { return m_external_psk_identity; }
 
+      /**
+       * Indicates that the session was established using an externally provided
+       * PSK. Session resumptions in TLS 1.3 (while technically implemented
+       * using a PSK) are not considered here. @sa was_resumption()
+       *
+       * @returns true if the session was established using an externally
+       *          provided PSK.
+       */
+      bool psk_used() const { return m_external_psk_identity.has_value(); }
```

Comment:
Note: With Botan 3.0 this reported `true` when a resumption was performed. This was technically correct, but semantically debatable at least. Now, `psk_used()` reports `true` for externally provided PSKs only.